UNITTESTS/drivers/Watchdog/test_watchdog.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2018, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "drivers/Watchdog.h"
+
+class TestWatchdog : public testing::Test {
+public:
+    static uint32_t expect_assert_count;
+    static uint32_t expect_reset_count;
+protected:
+    virtual void SetUp() {}
+    virtual void TearDown() {}
+};
+
+uint32_t TestWatchdog::expect_assert_count = 0;
+uint32_t TestWatchdog::expect_reset_count = 0;
+
+void mbed_assert_internal(const char *expr, const char *file, int line)
+{
+    TestWatchdog::expect_assert_count++;
+}
+
+void mock_system_reset()
+{
+    TestWatchdog::expect_reset_count++;
+}
+
+TEST_F(TestWatchdog, wdog_constructor)
+{
+    EXPECT_LE(sizeof(mbed::Watchdog), 1024);
+    mbed::Watchdog watchdog(500, "watchdog_unittest");
+}
+
+TEST_F(TestWatchdog, wdog_constructor_with_default_value)
+{
+    mbed::Watchdog watchdog;
```

Comment:
These constructor tests don't test anything.

Reply:
This is to make sure able to create watchdog instance without parameter.